cmd/build.go

```diff
@@ -41,45 +42,70 @@ func Build(ctx context.Context) *cobra.Command {
 				return err
 			}
 
-			path := "."
+			service := ""
 			if len(args) == 1 {
-				path = args[0]
+				service = args[0]
 			}
 
-			if err := utils.CheckIfDirectory(path); err != nil {
-				return fmt.Errorf("invalid build context: %s", err.Error())
+			manifestPath := "."
+			if options.File != "" {
+				if err := utils.CheckIfRegularFile(options.File); err != nil {
+					return fmt.Errorf("invalid File: %s", err.Error())
+				}
+				manifestPath = options.File
+			} else {
+				manifestPath = contextCMD.GetOktetoManifestPath(manifestPath)
 			}
-			options.Path = path
+			if manifestPath != "" {
```

Comment:
These changes seem long and a bit nested? Is it worth pulling it out into a function and writing a test to make sure we hit all the correct conditionals created when we expect to?